src/axom/quest/MarchingCubesAlgo.cpp

```diff
@@ -0,0 +1,614 @@
+{
+  _fcnPath = "fields/" + fcnField;
+  SLIC_ASSERT(_dom->has_path(_fcnPath));
+  SLIC_ASSERT(_dom->fetch_existing(_fcnPath + "/association").as_string() ==
+              "vertex");
+  SLIC_ASSERT(_dom->has_path(_fcnPath + "/values"));
+}
+
+/*!
+  @brief Set the output surface mesh object.
+*/
+void MarchingCubesAlgo1::set_output_mesh(axom::mint::Mesh* surfaceMesh)
+{
+  _surfaceMesh = surfaceMesh;
+
+  if(_ndim == 2)
+  {
+    using SegmentMesh = axom::mint::UnstructuredMesh<axom::mint::SINGLE_SHAPE>;
+    SegmentMesh* mesh = dynamic_cast<SegmentMesh*>(_surfaceMesh);
+    SLIC_ASSERT_MSG(mesh, "Surface mesh for 2D problem must be a SegmentMesh");
+  }
+  else
+  {
+    using TriangleMesh = axom::mint::UnstructuredMesh<axom::mint::SINGLE_SHAPE>;
+    TriangleMesh* mesh = dynamic_cast<TriangleMesh*>(_surfaceMesh);
+    SLIC_ASSERT_MSG(mesh, "Surface mesh for 3D problem must be a TriangleMesh");
+  }
+}
+
+void MarchingCubesAlgo1::compute_iso_surface(double contourVal)
+{
+  SLIC_ASSERT_MSG(_surfaceMesh,
+                  "You must call set_output_mesh before compute_iso_surface.");
+  SLIC_ASSERT_MSG(
+    !_fcnPath.empty(),
+    "You must call set_function_field before compute_iso_surface.");
+
+  /*
+    Notes: how to handle non-interleaved coordinates.
+    We should be able to handle any blueprint-compatible domains.
+
+    Need an array interface that supports strides.
+
+    conduit DataArray uses DataType, which has strides.  But I'm not convinced
+    that it actually uses those strides to compute offsets.  Need to do a practice
+    problem or try it here.
+
+    axom::ArrayView doesn't have strides.  It can support multicomponent arrays
+    so I can represent an Nx3 or 3xN layout.  However, I would have to switch
+    point index and dimension index depending on which layout.  Won't work.
+  */
+
+  const conduit::Node& coordValues =
+    _dom->fetch_existing(_coordsetPath + "/values");
+  bool isInterleaved = conduit::blueprint::mcarray::is_interleaved(coordValues);
+  const int coordSp = isInterleaved ? _ndim : 1;
+  const double* xPtr = coordValues["x"].as_double_ptr();
+  const double* yPtr = _ndim >= 2 ? coordValues["y"].as_double_ptr() : nullptr;
+  const double* zPtr = _ndim >= 3 ? coordValues["z"].as_double_ptr() : nullptr;
+
+  auto& fcnValues = _dom->fetch_existing(_fcnPath + "/values");
+  const double* fcnPtr = fcnValues.as_double_ptr();
+
+  const int* maskPtr = nullptr;
+  if(!_maskPath.empty())
+  {
+    auto& maskValues = _dom->fetch_existing(_maskPath + "/values");
+    maskPtr = maskValues.as_int_ptr();
+  }
+
+  if(!_cellIdField.empty() &&
+     !_surfaceMesh->hasField(_cellIdField, axom::mint::CELL_CENTERED))
+  {
+    _surfaceMesh->createField<int>(_cellIdField, axom::mint::CELL_CENTERED);
+  }
+
+  _contourVal = contourVal;
+
+  if(_ndim == 2)
+  {
+    /*
+      For now, assume zero offsets and zero ghost width.
+      Eventually, we'll have to support index offsets to
+      handle data with ghosts.
```

Comment:
Please create issues to track the follow-up tasks for MC

Reply:
Will do.